src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.cs

```diff
@@ -510,9 +503,51 @@ private bool IsFirstTokenOnLine(TokenData tokenData1, TokenData tokenData2)
 
         private int GetTokenIndexInStream(SyntaxToken token)
         {
-            if (_tokenToIndexMap.TryGetValue(token, out var value))
+            var tokenIndex = _tokens.BinarySearch(token, TokenOrderComparer.Instance);
+            if (tokenIndex < 0)
+            {
+                return -1;
+            }
+
+            // Source characters cannot be assigned to multiple tokens. If the token has non-zero width, it will be an
+            // exact match for at most one token.
+            if (!token.FullSpan.IsEmpty)
             {
-                return value;
+                return _tokens[tokenIndex] == token ? tokenIndex : -1;
```

Comment:
How does this code ever end up with a token that is at the same start location but is from a different tree entirely? It seems like that's playing very fast and loose with tokens in a way I wouldn't have expected.

Reply:
➡️ `token` is a zero-width token which immediately precedes the first token in the `TokenStream`. Whether it's the case of a separate tree or zero-width tokens, this case was handled by the previous code and I was just being careful to preserve that in the new code.